### ISSUE TITLE
Branch selection in dropdown of Codemarks section should automatically pick up branch change

### ISSUE THREAD

**Issue by angel200sdnot** [TeamCodeStream/codestream#537]
After opening the codestream extension in visual studio code, it doesn't seem to properly update when changing branches

**Comment by planteater** [comment]
@angel200sdnot - I think I might need a little more clarification on this one. Here's what I tried, but let me know if this isn't what you were describing...

1. I'm checked out to branch "master" and create a codemark. When I view that codemark, it's properly identified as being created against master.
2. I change to branch "foo" from the command line.
3. I create another codemark and it is properly identified as beign created against foo.

**Comment by angel200sdnot** [comment]
@planteater Codemarks are properly tagged for different branches, the issue is switching between branches doesn't update the displayed codemarks. To fix it i have to reload visual studio code

https://user-images.githubusercontent.com/24626009/116888341-a19e2800-ac2b-11eb-9e6b-bcf06ffa5778.mp4

**Comment by angel200sdnot** [comment]
@planteater Opening a file will also fix the issue, if this is the intended behavior then the issue can be closed. I am not sure why it wasn't updating for me before.

**Comment by planteater** [comment]
@angel200sdnot - Unfortunately I wasn't able to play that video. Upload seems to be corrupted.

I should note, though, that the display of codemarks isn't tied to what branch you're checked out to. If you're on master and a teammate creates a codemark against feature/foo, you will still see that codemark in your version of the file. Of course, your version of the code may differ from your teammates, in which case when you view the codemark you'll see a diff.

In short, CodeStream always shows all codemarks in a given file, regardless of source branch. When you view the codemark we let you know what branch it was created against and let you know if the code doesn't match what you have locally.

**Comment by angel200sdnot** [comment]
@planteater I have uploaded a video to show current behavior, here you can see how codemarks gets updated when opening a file https://imgur.com/a/67SG77R

The behavior you mention only happens when codemarks is set to "Current repository"

Since it updates when switching files i will close this issue, sorry for the prematurely issue report and thanks for your help

**Comment by planteater** [comment]
Thanks for the video @angel200sdnot! Now I see the issue. That dropdown in the Codemarks section could definitely reflect the branch change more instantaneously. I'll get that change onto our roadmap and update you here once it's done.

**Comment by planteater** [comment]
Fixed in v11.0.8.
